Drilldowns in examples

streamich opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/58027

This PR adds developer examples of *Drilldown Manager* usage outside of Dashboard app.

```
yarn start --run-examples
```

go to "Developer examples" -> "UI Actions Enhanced" example plugin.

![image](https://user-images.githubusercontent.com/9773803/94044547-969a3400-fdce-11ea-826a-cbd0773a4000.png)

- Refactors Dashboard-to-Dashboard drilldown such that it can be used in other places.
- Adds example plugin for `ui_actions_enhanced` plugin.
- "Without embeddable example" shows how Drilldown Manager could be used in a setting without embeddables and very lean trigger context, such is ML app use case.
- "Without embeddable example (single button)" shows how the same button could be reused for Drilldown Manager actions as well as drilldown dynamic actions, could be useful for Canvas, if they decide to add triggers to elements.
- "With embeddable example" shows that URL drilldown could be re-used in other apps if they use embeddables and `VALUE_CLICK_TRIGGER`.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/)
* Commit: 334e26efcc33e5d8f5efa91f87778117c2b81457
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/execution/node/124/log)
* [Build OSS Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/execution/node/122/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/execution/node/136/log)

### Build metrics
> ‼️ metrics were not reported for [[#75640](https://github.com/elastic/kibana/pull/75640)@[334e26e](https://github.com/elastic/kibana/commit/334e26efcc33e5d8f5efa91f87778117c2b81457)]

### History
* :broken_heart: [Build #69650](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69650/) failed 0c6ebb454aaeace50b22989bde4df968c383ff39
* :broken_heart: [Build #69648](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69648/) failed 3399275e79f7e0180741738d2caf64fe95f9e249

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69648/","number":"69648","commit":"3399275e79f7e0180741738d2caf64fe95f9e249"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69650/","number":"69650","commit":"0c6ebb454aaeace50b22989bde4df968c383ff39"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/","number":"70781","commit":"334e26efcc33e5d8f5efa91f87778117c2b81457"}]}
PIPELINE-->

streamich commented:
@elasticmachine merge upstream

elasticmachine commented:
merge conflict between base and head

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/)
* Commit: 8cc1723bce3e4c4727fb2599cd6a710c784d506e
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build OSS Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/execution/node/122/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/execution/node/136/log)

### Build metrics
> ‼️ metrics were not reported for [[#75640](https://github.com/elastic/kibana/pull/75640)@[8cc1723](https://github.com/elastic/kibana/commit/8cc1723bce3e4c4727fb2599cd6a710c784d506e)]

### History
* :broken_heart: [Build #73750](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73750/) failed e1f90953e4ee6691a23bde05f1b9b1dc99ce6787
* :broken_heart: [Build #70781](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/) failed 334e26efcc33e5d8f5efa91f87778117c2b81457
* :broken_heart: [Build #69650](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69650/) failed 0c6ebb454aaeace50b22989bde4df968c383ff39
* :broken_heart: [Build #69648](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69648/) failed 3399275e79f7e0180741738d2caf64fe95f9e249

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69648/","number":"69648","commit":"3399275e79f7e0180741738d2caf64fe95f9e249"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69650/","number":"69650","commit":"0c6ebb454aaeace50b22989bde4df968c383ff39"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/","number":"70781","commit":"334e26efcc33e5d8f5efa91f87778117c2b81457"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73750/","number":"73750","commit":"e1f90953e4ee6691a23bde05f1b9b1dc99ce6787"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/","number":"74828","commit":"8cc1723bce3e4c4727fb2599cd6a710c784d506e"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/)
* Commit: cf02239a1da6cbafbc7edf3deac45662e95c20e8
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build OSS Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/execution/node/122/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/execution/node/136/log)

### Build metrics
> ‼️ metrics were not reported for [[#75640](https://github.com/elastic/kibana/pull/75640)@[cf02239](https://github.com/elastic/kibana/commit/cf02239a1da6cbafbc7edf3deac45662e95c20e8)]

### History
* :broken_heart: [Build #74828](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/) failed 8cc1723bce3e4c4727fb2599cd6a710c784d506e
* :broken_heart: [Build #73750](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73750/) failed e1f90953e4ee6691a23bde05f1b9b1dc99ce6787
* :broken_heart: [Build #70781](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/) failed 334e26efcc33e5d8f5efa91f87778117c2b81457
* :broken_heart: [Build #69650](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69650/) failed 0c6ebb454aaeace50b22989bde4df968c383ff39
* :broken_heart: [Build #69648](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69648/) failed 3399275e79f7e0180741738d2caf64fe95f9e249

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69648/","number":"69648","commit":"3399275e79f7e0180741738d2caf64fe95f9e249"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69650/","number":"69650","commit":"0c6ebb454aaeace50b22989bde4df968c383ff39"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/","number":"70781","commit":"334e26efcc33e5d8f5efa91f87778117c2b81457"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73750/","number":"73750","commit":"e1f90953e4ee6691a23bde05f1b9b1dc99ce6787"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/","number":"74828","commit":"8cc1723bce3e4c4727fb2599cd6a710c784d506e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/","number":"75570","commit":"cf02239a1da6cbafbc7edf3deac45662e95c20e8"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-app-arch (Team:AppArch)

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/)
* Commit: 9175ab989e6434bd7136d461296f9813246b924a
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build OSS Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/execution/node/122/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/execution/node/136/log)

### Build metrics
> ‼️ metrics were not reported for [[#75640](https://github.com/elastic/kibana/pull/75640)@[9175ab9](https://github.com/elastic/kibana/commit/9175ab989e6434bd7136d461296f9813246b924a)]

### History
* :broken_heart: [Build #75570](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/) failed cf02239a1da6cbafbc7edf3deac45662e95c20e8
* :broken_heart: [Build #74828](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/) failed 8cc1723bce3e4c4727fb2599cd6a710c784d506e
* :broken_heart: [Build #73750](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73750/) failed e1f90953e4ee6691a23bde05f1b9b1dc99ce6787
* :broken_heart: [Build #70781](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/) failed 334e26efcc33e5d8f5efa91f87778117c2b81457
* :broken_heart: [Build #69650](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69650/) failed 0c6ebb454aaeace50b22989bde4df968c383ff39

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69650/","number":"69650","commit":"0c6ebb454aaeace50b22989bde4df968c383ff39"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/","number":"70781","commit":"334e26efcc33e5d8f5efa91f87778117c2b81457"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73750/","number":"73750","commit":"e1f90953e4ee6691a23bde05f1b9b1dc99ce6787"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/","number":"74828","commit":"8cc1723bce3e4c4727fb2599cd6a710c784d506e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/","number":"75570","commit":"cf02239a1da6cbafbc7edf3deac45662e95c20e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/","number":"76602","commit":"9175ab989e6434bd7136d461296f9813246b924a"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/)
* Commit: f19591f06ce18813106a30875c3acd10cc48cb00
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build OSS Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/execution/node/122/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/execution/node/136/log)

### Build metrics
> ‼️ metrics were not reported for [[#75640](https://github.com/elastic/kibana/pull/75640)@[f19591f](https://github.com/elastic/kibana/commit/f19591f06ce18813106a30875c3acd10cc48cb00)]

### History
* :broken_heart: [Build #76602](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/) failed 9175ab989e6434bd7136d461296f9813246b924a
* :broken_heart: [Build #75570](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/) failed cf02239a1da6cbafbc7edf3deac45662e95c20e8
* :broken_heart: [Build #74828](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/) failed 8cc1723bce3e4c4727fb2599cd6a710c784d506e
* :broken_heart: [Build #73750](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73750/) failed e1f90953e4ee6691a23bde05f1b9b1dc99ce6787
* :broken_heart: [Build #70781](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/) failed 334e26efcc33e5d8f5efa91f87778117c2b81457

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70781/","number":"70781","commit":"334e26efcc33e5d8f5efa91f87778117c2b81457"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73750/","number":"73750","commit":"e1f90953e4ee6691a23bde05f1b9b1dc99ce6787"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/","number":"74828","commit":"8cc1723bce3e4c4727fb2599cd6a710c784d506e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/","number":"75570","commit":"cf02239a1da6cbafbc7edf3deac45662e95c20e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/","number":"76602","commit":"9175ab989e6434bd7136d461296f9813246b924a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/","number":"76610","commit":"f19591f06ce18813106a30875c3acd10cc48cb00"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/)
* Commit: f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build OSS Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/execution/node/122/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/execution/node/136/log)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ‼️ metrics were not reported for [[#75640](https://github.com/elastic/kibana/pull/75640)@[f5c550b](https://github.com/elastic/kibana/commit/f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a)]

### History
* :broken_heart: [Build #76610](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/) failed f19591f06ce18813106a30875c3acd10cc48cb00
* :broken_heart: [Build #76602](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/) failed 9175ab989e6434bd7136d461296f9813246b924a
* :broken_heart: [Build #75570](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/) failed cf02239a1da6cbafbc7edf3deac45662e95c20e8
* :broken_heart: [Build #74828](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/) failed 8cc1723bce3e4c4727fb2599cd6a710c784d506e
* :broken_heart: [Build #73750](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73750/) failed e1f90953e4ee6691a23bde05f1b9b1dc99ce6787

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73750/","number":"73750","commit":"e1f90953e4ee6691a23bde05f1b9b1dc99ce6787"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/","number":"74828","commit":"8cc1723bce3e4c4727fb2599cd6a710c784d506e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/","number":"75570","commit":"cf02239a1da6cbafbc7edf3deac45662e95c20e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/","number":"76602","commit":"9175ab989e6434bd7136d461296f9813246b924a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/","number":"76610","commit":"f19591f06ce18813106a30875c3acd10cc48cb00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/","number":"76900","commit":"f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a"}]}
PIPELINE-->

Dosant commented on x-pack/examples/ui_actions_enhanced_examples/public/drilldowns/canvas_to_dashboard_drilldown/canvas_to_dashboard_drilldown.ts:
```diff
@@ -0,0 +1,35 @@
+type Trigger = typeof SAMPLE_CANVAS_ELEMENT_CLICK_TRIGGER;
+type Context = SampleCanvasElementClickContext;
+
+export class SampleCanvasToDashboardDrilldown extends AbstractDashboardDrilldown<Trigger> {
```
For me personally it is not easy to follow this pattern with abstract factory. I know we already introduced similar with `expore_underlying_data` and I didn't like it there earthier. 
I'd prefer just extracting common logic into functions and reusing them. For me it is easier to follow even though it leads to a bit more code.

In an intention of "one way of doing things", do you think it worth brining this up to wider team for discussion?

streamich replied:
The idea is simple, you extend that class and all you need to do is implement is the `getURL` function, and you have Dashboard drilldown working in your app.

Before arriving at this approach, I've tried a couple other options, including just creating a separate drilldown and reusing the React components of Dashboard drilldown and config functions, but that created a lot of messy imports, thus settled for this approach. With this approach it is only 15 lines of code to define a dashboard drilldown in your app.

But curious to see if you have in mind a nicer way to do it. If you have in mind making all the functions like `execute`, `CollectConfig`, `validateConfig`, `isCompatible`, `getDisplayName`, `getIconType` as standalone, which can be imported by multiple drilldowns, I would not like it over the current solution; as it is one of the options I tried and it was messy to track all the imports. Here you really need to import just one thing, i.e. `AbstractDashboardDrilldown` and extend it.

Dosant commented on x-pack/plugins/dashboard_enhanced/public/services/drilldowns/abstract_dashboard_drilldown/abstract_dashboard_drilldown.tsx:
```diff
@@ -0,0 +1,89 @@
+
+  public readonly execute = async (config: Config, context: TriggerContextMapping[T]) => {
+    const url = await this.getURL(config, context);
+    await this.params.start().core.application.navigateToApp(url.appName, { path: url.appPath });
```
Btw, there is also `navigateToUrl` method. 
I think we can and should use it. Then we don't need URL parsing and can remove temporary `KibanaURL` implementation. 

```
this.params.start().core.application.navigateToUrl(await this.getURL(config, context))
```

streamich replied:
Yes, good catch, I've also seen it while refactoring this, but decided to leave as is for now so to not accidentally break anything.

Dosant commented on x-pack/plugins/ui_actions_enhanced/public/drilldowns/components/drilldown_hello_bar/drilldown_hello_bar.tsx:
```diff
@@ -31,31 +31,31 @@ export const DrilldownHelloBar: React.FC<DrilldownHelloBarProps> = ({
-      <EuiFlexGroup>
-        <EuiFlexItem grow={false}>
-          <div style={{ marginLeft: '8px' }}>
+      <div style={{ marginBottom: -12 }}>
```
`marginBottom: -12` - this doesn't do anything. there is !important in `eui` styles.
Not sure why changing this component was needed also

Dosant replied:
With this change the vertical alignment between help icon and help text got worse: 

![94445467-c28b3000-01a7-11eb-9db0-d082b6ddfda8](https://user-images.githubusercontent.com/7784120/94450769-f2d5cd00-01ad-11eb-9905-be11313344c8.png)

Can we fix it?

streamich replied:
I tried a bit, but could not find a nice way, probably better ask EUI to fix it.

Dosant replied:
If to pick between first and second image, I prefer 😅. 
Let's revert and create EUI issue? :D

streamich replied:
I think the second image where icon is aligned with the title and where top and bottom paddings are equal looks much better. Also I think the code of the second image is better because it gets rid of `style={{ marginLeft: '8px' }}`.

I suggest we keep the current change and leave it up to @elastic/eui-design if they want to move the icon lower or if they are OK with how it looks now.

Dosant replied:
A heads up: will conflict https://github.com/elastic/kibana/pull/78801/files#diff-a247aa12e5b9e7ad760fe6c5880fc7efL36

cchaos replied:
This is a completely custom implementation of the content for EuiCallout, so I'm not seeing what we can do on the EUI side. The ideal vertical alignment would be for the icon to either sit at the baseline of the text or centered with the first line of text not at the top of the letter. But the version without the bottom margin looks just fine to me.

Dosant commented on x-pack/plugins/ui_actions_enhanced/public/drilldowns/components/connected_flyout_manage_drilldowns/connected_flyout_manage_drilldowns.tsx:
```diff
@@ -250,108 +242,3 @@ export function createFlyoutManageDrilldowns({
   };
 }
-
-function useCompatibleActionFactoriesForCurrentContext<
```
didn't see a reason moving it to separate file :D. enjoyed how it's all centralised. :(
aren't asking to get it back.. just saying.

streamich replied:
The `connected_flyout_manage_drilldowns.tsx` file is growing too large and I found confusing to navigate in it, especially when React hooks are in the same file and are defined after the definition of component which uses them.

I think we should further clean up this file, ideally it would be just two lines like this:

```tsx
    if (route ===  Routes.Manage) return <ConnectedFlyoutListManageDrilldowns />;
    return <ConnectedFlyoutDrilldownWizard />;
```

instead of this *SwitchStatement*:

```tsx

    switch (route) {
      case Routes.Create:
      case Routes.Edit:
        return (
          <FlyoutDrilldownWizard
            docsLink={docsLink}
            triggerPickerDocsLink={triggerPickerDocsLink}
            showWelcomeMessage={shouldShowWelcomeMessage}
            onWelcomeHideClick={onHideWelcomeMessage}
            drilldownActionFactories={actionFactories}
            onClose={props.onClose}
            mode={route === Routes.Create ? 'create' : 'edit'}
            onBack={isCreateOnly ? undefined : () => setRoute(Routes.Manage)}
            onSubmit={({ actionConfig, actionFactory, name, selectedTriggers }) => {
              if (route === Routes.Create) {
                createDrilldown(
                  {
                    name,
                    config: actionConfig,
                    factoryId: actionFactory.id,
                  },
                  selectedTriggers
                );
              } else {
                editDrilldown(
                  currentEditId!,
                  {
                    name,
                    config: actionConfig,
                    factoryId: actionFactory.id,
                  },
                  selectedTriggers
                );
              }

              if (isCreateOnly) {
                if (props.onClose) {
                  props.onClose();
                }
              } else {
                setRoute(Routes.Manage);
              }

              setCurrentEditId(null);
            }}
            onDelete={() => {
              deleteDrilldown(currentEditId!);
              setRoute(Routes.Manage);
              setCurrentEditId(null);
            }}
            actionFactoryPlaceContext={props.placeContext}
            initialDrilldownWizardConfig={resolveInitialDrilldownWizardConfig()}
            supportedTriggers={props.triggers}
            getTrigger={getTrigger}
          />
        );

      case Routes.Manage:
      default:
        // show trigger column in case if there is more then 1 possible trigger in current context
        const showTriggerColumn =
          intersection(
            props.triggers,
            actionFactories
              .map((factory) => factory.supportedTriggers())
              .reduce((res, next) => res.concat(next), [])
          ).length > 1;
        return (
          <FlyoutListManageDrilldowns
            docsLink={docsLink}
            showWelcomeMessage={shouldShowWelcomeMessage}
            onWelcomeHideClick={onHideWelcomeMessage}
            drilldowns={drilldowns.map(mapToDrilldownToDrilldownListItem)}
            onDelete={(ids) => {
              setCurrentEditId(null);
              deleteDrilldown(ids);
            }}
            onEdit={(id) => {
              setCurrentEditId(id);
              setRoute(Routes.Edit);
            }}
            onCreate={() => {
              setCurrentEditId(null);
              setRoute(Routes.Create);
            }}
            onClose={props.onClose}
            showTriggerColumn={showTriggerColumn}
          />
        );
    }
```

Dosant reviewed (commented):
This all is really cool and looks great. 

Some suggestions: 

1. In code you created fake "canvas" and "ml" contexts and actions. But then the application UI doesn't refer to them as "canvas" or "ml". This was very confusing to me to explore code after playing with the app and see references to "ml". I'd suggest to replace those with "sampleApp1" and "sampleApp2"? I think if we want to focus on "your app can have it's own triggers and drilldown types" then we also should explain it in examples descriptions on UI and make it clearer inside the drilldowns and triggers display texts. If we don't want to focus on that, then I'd just simplify to using our existing triggers. 

2. Sample URL Drilldown is copied as a sample app specific drilldown type. This wasn't clear from the example app that those drilldown types in different examples are actually different types. I'd suggest to simplify these examples to hello world samples and make it clear in the messaging on UI and in alert execution that those are actually different app specific types? 

3. Original _dashboard_drilldown_ implementation is not used, so old code should be removed?

streamich commented on x-pack/examples/ui_actions_enhanced_examples/public/containers/drilldowns_with_embeddable_example/drilldowns_with_embeddable_example.tsx:
```diff
@@ -0,0 +1,134 @@
+      </EuiFlexGroup>
+
+      {showManager && (
+        <EuiFlyout onClose={() => setShowManager(false)} aria-labelledby="Drilldown Manager">
```
Thanks, I've actually tried it but something is wrong with overlays in Kibana:

![image](https://user-images.githubusercontent.com/9773803/94442377-2dd30300-01a4-11eb-8712-4525077d5251.png)

Dosant replied:
Maybe that's why we use core's overlay service for all overlays? And not just render flouts as you did in this pr 🤔

streamich replied:
Flyout from core is useful when don't have a React tree where to put EUI flyout. If you are using React, it is better to use EUI flyout.

streamich replied:
Also, I believe we use both approaches throughout code.

streamich commented on x-pack/examples/ui_actions_enhanced_examples/public/containers/drilldowns_with_embeddable_example/drilldowns_with_embeddable_example.tsx:
```diff
@@ -0,0 +1,134 @@
+
+      {showManager && (
+        <EuiFlyout onClose={() => setShowManager(false)} aria-labelledby="Drilldown Manager">
+          <plugins.uiActionsEnhanced.FlyoutManageDrilldowns
```
Looks like ti is coming from "say hello" sample drilldown.

streamich replied:
Ah, no, it was actually coming from "Get more actions" text, fixed.

streamich commented:
> 1. In code you created fake "canvas" and "ml" contexts and actions. But then the application UI doesn't refer to them as "canvas" or "ml". This was very confusing to me to explore code after playing with the app and see references to "ml". I'd suggest to replace those with "sampleApp1" and "sampleApp2"? [..]

Done. Yes, agreed, it was confusing to me while creating those as well :smile:

> 2. [..] I'd suggest to simplify these examples to hello world samples and make it clear in the messaging on UI and in alert execution that those are actually different app specific types?

I like this suggestion, done.

> 3. Original dashboard_drilldown implementation is not used, so old code should be removed?

Yes, good catch, forgot to remove that last file.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/)
* Commit: dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/execution/node/380/log)

### History
* :broken_heart: [Build #76900](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/) failed f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a
* :broken_heart: [Build #76610](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/) failed f19591f06ce18813106a30875c3acd10cc48cb00
* :broken_heart: [Build #76602](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/) failed 9175ab989e6434bd7136d461296f9813246b924a
* :broken_heart: [Build #75570](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/) failed cf02239a1da6cbafbc7edf3deac45662e95c20e8
* :broken_heart: [Build #74828](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/) failed 8cc1723bce3e4c4727fb2599cd6a710c784d506e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74828/","number":"74828","commit":"8cc1723bce3e4c4727fb2599cd6a710c784d506e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/","number":"75570","commit":"cf02239a1da6cbafbc7edf3deac45662e95c20e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/","number":"76602","commit":"9175ab989e6434bd7136d461296f9813246b924a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/","number":"76610","commit":"f19591f06ce18813106a30875c3acd10cc48cb00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/","number":"76900","commit":"f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/","number":"77611","commit":"dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4"}]}
PIPELINE-->

streamich commented:
@elasticmachine merge upstream

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77753/)
* Commit: 08dd6dbe8f45de9864d7ab206818c156b987acfd
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77753/execution/node/358/log)

### History
* :broken_heart: [Build #77611](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/) failed dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4
* :broken_heart: [Build #76900](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/) failed f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a
* :broken_heart: [Build #76610](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/) failed f19591f06ce18813106a30875c3acd10cc48cb00
* :broken_heart: [Build #76602](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/) failed 9175ab989e6434bd7136d461296f9813246b924a
* :broken_heart: [Build #75570](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/) failed cf02239a1da6cbafbc7edf3deac45662e95c20e8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75570/","number":"75570","commit":"cf02239a1da6cbafbc7edf3deac45662e95c20e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/","number":"76602","commit":"9175ab989e6434bd7136d461296f9813246b924a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/","number":"76610","commit":"f19591f06ce18813106a30875c3acd10cc48cb00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/","number":"76900","commit":"f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/","number":"77611","commit":"dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77753/","number":"77753","commit":"08dd6dbe8f45de9864d7ab206818c156b987acfd"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77850/)
* Commit: d848d29aa91e2f80ab0e61a03fa6af5fc2caa6da
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77850/execution/node/362/log)

### History
* :broken_heart: [Build #77753](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77753/) failed 08dd6dbe8f45de9864d7ab206818c156b987acfd
* :broken_heart: [Build #77611](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/) failed dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4
* :broken_heart: [Build #76900](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/) failed f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a
* :broken_heart: [Build #76610](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/) failed f19591f06ce18813106a30875c3acd10cc48cb00
* :broken_heart: [Build #76602](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/) failed 9175ab989e6434bd7136d461296f9813246b924a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76602/","number":"76602","commit":"9175ab989e6434bd7136d461296f9813246b924a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/","number":"76610","commit":"f19591f06ce18813106a30875c3acd10cc48cb00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/","number":"76900","commit":"f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/","number":"77611","commit":"dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77753/","number":"77753","commit":"08dd6dbe8f45de9864d7ab206818c156b987acfd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77850/","number":"77850","commit":"d848d29aa91e2f80ab0e61a03fa6af5fc2caa6da"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/)
* Commit: 83701c0764a7b065a667ebe7452d8094578ae3c1
* This comment will update when the build is complete

### Failed CI Steps
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/execution/node/468/log)
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/execution/node/577/log)

### History
* :broken_heart: [Build #77850](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77850/) failed d848d29aa91e2f80ab0e61a03fa6af5fc2caa6da
* :broken_heart: [Build #77753](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77753/) failed 08dd6dbe8f45de9864d7ab206818c156b987acfd
* :broken_heart: [Build #77611](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/) failed dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4
* :broken_heart: [Build #76900](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/) failed f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a
* :broken_heart: [Build #76610](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/) failed f19591f06ce18813106a30875c3acd10cc48cb00

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76610/","number":"76610","commit":"f19591f06ce18813106a30875c3acd10cc48cb00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/","number":"76900","commit":"f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/","number":"77611","commit":"dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77753/","number":"77753","commit":"08dd6dbe8f45de9864d7ab206818c156b987acfd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77850/","number":"77850","commit":"d848d29aa91e2f80ab0e61a03fa6af5fc2caa6da"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/","number":"77939","commit":"83701c0764a7b065a667ebe7452d8094578ae3c1"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77967/)
* Commit: d04d7fce75871c4cb5c330ff0e41190af518f16e
* This comment will update when the build is complete

### Failed CI Steps
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77967/execution/node/468/log)
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77967/execution/node/577/log)

### History
* :broken_heart: [Build #77939](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/) failed 83701c0764a7b065a667ebe7452d8094578ae3c1
* :broken_heart: [Build #77850](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77850/) failed d848d29aa91e2f80ab0e61a03fa6af5fc2caa6da
* :broken_heart: [Build #77753](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77753/) failed 08dd6dbe8f45de9864d7ab206818c156b987acfd
* :broken_heart: [Build #77611](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/) failed dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4
* :broken_heart: [Build #76900](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/) failed f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/76900/","number":"76900","commit":"f5c550bbf2678c57e0765d7cb22fd05fbfcfbd5a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/","number":"77611","commit":"dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77753/","number":"77753","commit":"08dd6dbe8f45de9864d7ab206818c156b987acfd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77850/","number":"77850","commit":"d848d29aa91e2f80ab0e61a03fa6af5fc2caa6da"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/","number":"77939","commit":"83701c0764a7b065a667ebe7452d8094578ae3c1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77967/","number":"77967","commit":"d04d7fce75871c4cb5c330ff0e41190af518f16e"}]}
PIPELINE-->

Dosant commented on x-pack/examples/ui_actions_enhanced_examples/public/plugin.ts:
```diff
@@ -4,44 +4,174 @@
+              viewMode: 'create',
+              dynamicActionManager: self.managerWithoutEmbeddableSingleButton,
+              triggers: [SAMPLE_APP2_CLICK_TRIGGER],
+              placeContext: { sampleCanavsElementClickContext: sampleApp2ClickContext },
```
a typo

```suggestion
              placeContext: { sampleCanvasElementClickContext: sampleApp2ClickContext },
```

streamich replied:
Done.

Dosant commented on x-pack/examples/ui_actions_enhanced_examples/public/plugin.ts:
```diff
@@ -4,44 +4,174 @@
+              viewMode: 'create',
+              dynamicActionManager: self.managerWithoutEmbeddableSingleButton,
+              triggers: [SAMPLE_APP2_CLICK_TRIGGER],
+              placeContext: { sampleCanavsElementClickContext: sampleApp2ClickContext },
```
I also can't find where `sampleCanavsElementClickContext` if it is not used, let's remove to simplify examples? 
If it is used, let's completely go away from `canvas` `ml` references and use `dummy` references instead

streamich replied:
Done.

Dosant commented on x-pack/examples/ui_actions_enhanced_examples/public/triggers/sample_app2_trigger.ts:
```diff
@@ -0,0 +1,31 @@
+}
+
+export interface SampleApp2ClickContext {
+  workpadId: string;
```
Let's get rid of all `canvas` or `ml` references in examples?

streamich replied:
I think it is fine to have `workpadId` here, it does not need to be a reference to `canvas` or `ml`, just an example `workpadId`.

Dosant commented on x-pack/examples/ui_actions_enhanced_examples/public/triggers/sample_app1_trigger.ts:
```diff
@@ -0,0 +1,31 @@
+}
+
+export interface SampleApp1ClickContext {
+  job: SampleMlJob;
```
(repeat) Let's get rid of all canvas or ml references in examples?

streamich replied:
Same here, it is just a context of "app 1" and it shows that it can have `job` field (or any field).

Dosant commented on x-pack/examples/ui_actions_enhanced_examples/public/drilldowns/app2_to_dashboard_drilldown/app2_to_dashboard_drilldown.ts:
```diff
@@ -0,0 +1,32 @@
+  public readonly id = APP2_TO_DASHBOARD_DRILLDOWN;
+
+  public readonly supportedTriggers = () => [SAMPLE_APP2_CLICK_TRIGGER] as Trigger[];
+
```
Maybe let's overrides display name in these two. Instead of `Go to dashboard` make `Go to dashboard (app2)`

streamich replied:
No, the whole idea of the dashboard drilldown now is that it can be used in any app, I don't see why we would want to specify `(app2)`.

Dosant commented on x-pack/examples/ui_actions_enhanced_examples/public/drilldowns/dashboard_to_url_drilldown/index.tsx:
```diff
@@ -0,0 +1,142 @@
+
+const SAMPLE_DASHBOARD_TO_URL_DRILLDOWN = 'SAMPLE_DASHBOARD_TO_URL_DRILLDOWN';
+
+export class DashboardToUrlDrilldown implements Drilldown<Config, UrlTrigger> {
```
This Drilldown is not used? I deleted it with URL drilldown pr. I think we don't need to carry this around

streamich replied:
Nope, not used, deleted.

Dosant commented on x-pack/plugins/dashboard_enhanced/public/services/drilldowns/abstract_dashboard_drilldown/abstract_dashboard_drilldown.tsx:
```diff
@@ -0,0 +1,89 @@
+    CollectConfigProps<Config, BaseActionFactoryContext<T>>
+  > = (props) => <CollectConfigContainer {...props} params={this.params} />;
+
+  public readonly CollectConfig = reactToUiComponent(this.ReactCollectConfig);
```
Should we conditionally display filters / time range inside collect config depends if it available on embeddable input? 

These inputs don't make sense in example plugin:
![Screenshot 2020-09-30 at 12 26 21](https://user-images.githubusercontent.com/7784120/94674704-e02bd700-0318-11eb-9993-5d23fbe83780.png)

I am OK if we at least create an issue to clean this up.

streamich replied:
Good point, I've created an issue for it: https://github.com/elastic/kibana/issues/78997

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78221/)
* Commit: 8140373593cff7307f2f119fd5a6deb9f191af56
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Integration Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78221/execution/node/398/log)

### History
* :broken_heart: [Build #77967](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77967/) failed d04d7fce75871c4cb5c330ff0e41190af518f16e
* :broken_heart: [Build #77939](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/) failed 83701c0764a7b065a667ebe7452d8094578ae3c1
* :broken_heart: [Build #77850](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77850/) failed d848d29aa91e2f80ab0e61a03fa6af5fc2caa6da
* :broken_heart: [Build #77753](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77753/) failed 08dd6dbe8f45de9864d7ab206818c156b987acfd
* :broken_heart: [Build #77611](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/) failed dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77611/","number":"77611","commit":"dd2ce1d6a3ed9503a711e53cc6e7c9bc680d6ba4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77753/","number":"77753","commit":"08dd6dbe8f45de9864d7ab206818c156b987acfd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77850/","number":"77850","commit":"d848d29aa91e2f80ab0e61a03fa6af5fc2caa6da"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/","number":"77939","commit":"83701c0764a7b065a667ebe7452d8094578ae3c1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77967/","number":"77967","commit":"d04d7fce75871c4cb5c330ff0e41190af518f16e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78221/","number":"78221","commit":"8140373593cff7307f2f119fd5a6deb9f191af56"}]}
PIPELINE-->

Dosant commented on x-pack/plugins/ui_actions_enhanced/public/drilldowns/components/connected_flyout_manage_drilldowns/connected_flyout_manage_drilldowns.tsx:
```diff
@@ -4,33 +4,26 @@
-import useMountedState from 'react-use/lib/useMountedState';
-import { intersection } from 'lodash';
+// eslint-disable-next-line no-restricted-imports
+import intersection from 'lodash/intersection';
```
please revert (this adds more bytes into bundle)
explanation: https://github.com/elastic/kibana/pull/78156#issue-490969591

streamich replied:
Done.

Dosant commented on x-pack/plugins/ui_actions_enhanced/public/drilldowns/hooks/i18n.ts:
```diff
@@ -0,0 +1,88 @@
+    {
+      // TODO: remove `Save your dashboard before testing.` part
+      // when drilldowns are used not only in dashboard
+      // or after https://github.com/elastic/kibana/issues/65179 implemented
```
Should we just remove it? I doubt that it brings any value and removing it will make drilldown more generic :)

streamich replied:
Yes, I agree on removing it, but let's ask Jason if he is OK with it, too, I've added it to the sync agenda.

Dosant replied:
(From a meeting) "Save you changes before testing". We will remove after https://github.com/elastic/kibana/issues/65179 is done

Dosant reviewed (commented):
Looks good, 

1. I think removing `ml` / `canvas` reference in code is a right call, I just think we should finalise it as there are still references, for example, in trigger contexts shapes. 

2. The example app is so nice. I'd just think we should push descriptions / displayNames / labels a bit further. I don't think anyone would get an idea that apps can register their own triggers and create there own drilldown types for those triggers by just playing with the app. For example: it is not clear that we have different "go to dashboard" implementations from example app.

3. Minor nit, let's at least add description? 

![Screenshot 2020-09-30 at 12 25 23](https://user-images.githubusercontent.com/7784120/94675466-1158d700-031a-11eb-86a4-0ab3801c210b.png)


Some other asks below. Some of them intersect with points above

streamich commented:
@Dosant 

1. I've removed references, except the very deep ones inside the context object, which are not really references to ML or Canvas but just shows that the context can have app specific data.

2. Yes, I think for that we need to add explanations to the README, I've added explanation of what is contained in the sample folders.

3. I've added the description.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78616/)
* Commit: 0c5ac3be9bafdd5f714a0f129c7af819499bd048

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### @kbn/optimizer bundle module count
| id | [before](https://github.com/elastic/kibana/commit/0c5ac3be9bafdd5f714a0f129c7af819499bd048) | [after](https://github.com/elastic/kibana/commit/65cf6393c770af89567685980313e464e69745d4) | diff |
| --- | --- | --- | --- |
| `dashboardEnhanced` | 26 | 28 | +2 |
| `discoverEnhanced` | 11 | 10 | -1 |
| `share` | 38 | 39 | +1 |
| `uiActionsEnhanced` | 134 | 138 | +4 |
| total |  |  | +6 |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/0c5ac3be9bafdd5f714a0f129c7af819499bd048) | [after](https://github.com/elastic/kibana/commit/65cf6393c770af89567685980313e464e69745d4) | diff |
| --- | --- | --- | --- |
| `dashboard` | 570.4KB | 570.5KB | +60.0B |
| `dashboardEnhanced` | 43.1KB | 49.5KB | +6.4KB |
| `discoverEnhanced` | 29.2KB | 27.1KB | -2.1KB |
| `share` | 80.0KB | 82.2KB | +2.2KB |
| `uiActionsEnhanced` | 320.7KB | 326.7KB | +6.1KB |
| total |  |  | +12.6KB |

### History
* :broken_heart: [Build #78426](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78426/) failed ee6769d5197890c03cc76f7cdbab8b9b38c087bb
* :broken_heart: [Build #78221](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78221/) failed 8140373593cff7307f2f119fd5a6deb9f191af56
* :broken_heart: [Build #77967](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77967/) failed d04d7fce75871c4cb5c330ff0e41190af518f16e
* :broken_heart: [Build #77939](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/) failed 83701c0764a7b065a667ebe7452d8094578ae3c1
* :broken_heart: [Build #77850](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77850/) failed d848d29aa91e2f80ab0e61a03fa6af5fc2caa6da

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77850/","number":"77850","commit":"d848d29aa91e2f80ab0e61a03fa6af5fc2caa6da"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/","number":"77939","commit":"83701c0764a7b065a667ebe7452d8094578ae3c1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77967/","number":"77967","commit":"d04d7fce75871c4cb5c330ff0e41190af518f16e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78221/","number":"78221","commit":"8140373593cff7307f2f119fd5a6deb9f191af56"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78426/","number":"78426","commit":"ee6769d5197890c03cc76f7cdbab8b9b38c087bb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78616/","number":"78616","commit":"0c5ac3be9bafdd5f714a0f129c7af819499bd048"}]}
PIPELINE-->

streamich commented:
@elasticmachine merge upstream

Dosant reviewed (approved):
LGTM, didn't retest latest changes. 
One thing: https://github.com/elastic/kibana/pull/75640/files#r498278987 (note about `dashboard` word in toast)
Or I guess it is fine to change in scope of https://github.com/elastic/kibana/issues/78997

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79498/)
* Commit: 3eb5e278585c50beda3c4b9bb2b12bdaeb4e6366

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### @kbn/optimizer bundle module count
| id | [before](https://github.com/elastic/kibana/commit/3eb5e278585c50beda3c4b9bb2b12bdaeb4e6366) | [after](https://github.com/elastic/kibana/commit/9e1c44fcc08351070afffe87056cd86aefbe3449) | diff |
| --- | --- | --- | --- |
| `dashboardEnhanced` | 26 | 28 | +2 |
| `discoverEnhanced` | 11 | 10 | -1 |
| `share` | 38 | 39 | +1 |
| `uiActionsEnhanced` | 134 | 138 | +4 |
| total |  |  | +6 |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/3eb5e278585c50beda3c4b9bb2b12bdaeb4e6366) | [after](https://github.com/elastic/kibana/commit/9e1c44fcc08351070afffe87056cd86aefbe3449) | diff |
| --- | --- | --- | --- |
| `dashboard` | 570.2KB | 570.3KB | +60.0B |
| `dashboardEnhanced` | 43.1KB | 49.5KB | +6.4KB |
| `discoverEnhanced` | 29.2KB | 27.1KB | -2.1KB |
| `share` | 79.9KB | 82.1KB | +2.2KB |
| `uiActionsEnhanced` | 320.7KB | 326.8KB | +6.1KB |
| total |  |  | +12.6KB |

### History
* :green_heart: [Build #78616](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78616/) succeeded 0c5ac3be9bafdd5f714a0f129c7af819499bd048
* :broken_heart: [Build #78426](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78426/) failed ee6769d5197890c03cc76f7cdbab8b9b38c087bb
* :broken_heart: [Build #78221](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78221/) failed 8140373593cff7307f2f119fd5a6deb9f191af56
* :broken_heart: [Build #77967](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77967/) failed d04d7fce75871c4cb5c330ff0e41190af518f16e
* :broken_heart: [Build #77939](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/) failed 83701c0764a7b065a667ebe7452d8094578ae3c1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77939/","number":"77939","commit":"83701c0764a7b065a667ebe7452d8094578ae3c1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77967/","number":"77967","commit":"d04d7fce75871c4cb5c330ff0e41190af518f16e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78221/","number":"78221","commit":"8140373593cff7307f2f119fd5a6deb9f191af56"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78426/","number":"78426","commit":"ee6769d5197890c03cc76f7cdbab8b9b38c087bb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78616/","number":"78616","commit":"0c5ac3be9bafdd5f714a0f129c7af819499bd048"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79498/","number":"79498","commit":"3eb5e278585c50beda3c4b9bb2b12bdaeb4e6366"}]}
PIPELINE-->

majagrubic commented on x-pack/plugins/ui_actions_enhanced/public/drilldowns/hooks/use_drilldown_state_manager.tsx:
```diff
@@ -0,0 +1,86 @@
+      drilldownIds = Array.isArray(drilldownIds) ? drilldownIds : [drilldownIds];
+      await actionManager.deleteEvents(drilldownIds);
+      toastService.addSuccess(
+        drilldownIds.length === 1
```
It's generally a bad practice to have different texts based on a number. In some languages, there may be multiple forms of plurals, so not everything > 1 will behave the same. The recommended practice is to have a localization message that says something like: `Drilldowns deleted: {number}`

streamich replied:
I'll take a look into this in a follow up.

majagrubic commented on x-pack/plugins/ui_actions_enhanced/public/drilldowns/hooks/use_drilldown_state_manager.tsx:
```diff
@@ -0,0 +1,86 @@
+      });
+      if (!isMounted) return;
+      setIsLoading(false);
+      return;
```
This return is obsolete.

majagrubic reviewed (approved):
Had a look at the code, didn't run the examples. kibana-app code looks good to me 👍 
Just a few nits I noticed below 👇

streamich commented on x-pack/plugins/ui_actions_enhanced/public/drilldowns/hooks/use_compatible_action_factories_for_current_context.ts:
```diff
@@ -0,0 +1,35 @@
+>(actionFactories: ActionFactory[], context: Context) {
+  const [compatibleActionFactories, setCompatibleActionFactories] = useState<ActionFactory[]>();
+  useEffect(() => {
+    let canceled = false;
```
I'll fix it.
